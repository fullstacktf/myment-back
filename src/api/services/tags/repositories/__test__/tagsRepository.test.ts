import { TagsMongoRepository as TagsRepository } from '../TagsRepository';
describe('TagsRepository', () => {
    const repo = new TagsRepository();

    describe('getConnectionState', () => {
        test('Check connection to database', () => {
            return repo.getConnection().then(state => {
                expect(state.readyState).toBe(1);
            });
        });
    });
    describe('showLocations', () => {
        // test('should return all activities', () => {
        //     return repo.getAll().then(data => {
        //         expect(data).toBe('')
        //     });
        // });
        test('should return only Place EspaÃ±a',() => {
            return repo.showAll().then(data =>{
                expect(data).toMatchSnapshot();
            });
        })
        test('should return only Category Tags',() => {
            return repo.findByCategory('food').then(data =>{
                expect(data).toBe('');
            });
        })
    });
});
