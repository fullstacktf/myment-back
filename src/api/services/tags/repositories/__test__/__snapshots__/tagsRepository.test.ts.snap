// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsRepository showLocations should return only Place Espa√±a 1`] = `
Array [
  Object {
    "_id": "5d52e3b3727816af15e6de1a",
    "category": Array [
      "leisure",
      "lodging",
    ],
    "name": "Bonito",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de1b",
    "category": Array [
      "food",
    ],
    "name": "Rico",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de1c",
    "category": Array [
      "food",
    ],
    "name": "Sabroso",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de1d",
    "category": Array [
      "leisure",
    ],
    "name": "Aventura",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de1e",
    "category": Array [
      "leisure",
    ],
    "name": "Extremo",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de1f",
    "category": Array [
      "leisure",
      "lodging",
    ],
    "name": "Tranquilo",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de20",
    "category": Array [
      "lodging",
    ],
    "name": "Cuidado",
  },
  Object {
    "_id": "5d52e3b3727816af15e6de21",
    "category": Array [
      "leisure",
      "lodging",
    ],
    "name": "Bonito",
  },
]
`;
